interop-testing/src/main/java/io/grpc/testing/integration/GrpclbLongLivedAffinityTestClient.java

```diff
@@ -0,0 +1,218 @@
+    }
+  }
+
+  private String target = "directpath-grpclb-with-pick-first-test.googleapis.com";
+  private long rpcErrorBudgetIncreaseMinutes = 2;
+  private long affinityBreakageBudgetIncreaseMinutes = 90;
+  private long rpcIntermissionSeconds = 1;
+  private long totalTestSeconds = 60;
+
+  protected ManagedChannel channel;
+  protected TestServiceGrpc.TestServiceBlockingStub blockingStub;
+
+  private void parseArgs(String[] args) {
+    boolean usage = false;
+    for (String arg : args) {
+      if (!arg.startsWith("--")) {
+        System.err.println("All arguments must start with '--': " + arg);
+        usage = true;
+        break;
+      }
+      String[] parts = arg.substring(2).split("=", 2);
+      String key = parts[0];
+      if ("help".equals(key)) {
+        usage = true;
+        break;
+      }
+      if (parts.length != 2) {
+        System.err.println("All arguments must be of the form --arg=value");
+        usage = true;
+        break;
+      }
+      String value = parts[1];
+      if ("target".equals(key)) {
+        target = value;
+      } else if ("rpc_error_budget_increase_minutes".equals(key)) {
+        rpcErrorBudgetIncreaseMinutes = Long.parseLong(value);
+      } else if ("affinity_breakage_budget_increase_minutes".equals(key)) {
+        affinityBreakageBudgetIncreaseMinutes = Long.parseLong(value);
+      } else if ("rpc_intermission_seconds".equals(key)) {
+        rpcIntermissionSeconds = Long.parseLong(value);
+      } else if ("total_test_seconds".equals(key)) {
+        totalTestSeconds = Long.parseLong(value);
+      } else {
+        System.err.println("Unknown argument: " + key);
+        usage = true;
+        break;
+      }
+    }
+    if (usage) {
+      GrpclbLongLivedAffinityTestClient c = new GrpclbLongLivedAffinityTestClient();
+      System.out.println(
+          "Usage: [ARGS...]"
+          + "\n"
+          + "\n  --target=TARGET          Server target.             Default " + c.target
+          + "\n  --rpc_error_budget_increase_minutes=MINUTES         Default "
+          + c.rpcErrorBudgetIncreaseMinutes
+          + "\n  --affinity_breakage_budget_increase_minutes=MINUTES Default "
+          + c.affinityBreakageBudgetIncreaseMinutes
+          + "\n  --rpc_intermission_seconds=SECONDS                  Default "
+          + c.rpcIntermissionSeconds
+          + "\n  --total_test_seconds=SECONDS                        Default "
+          + c.totalTestSeconds
+      );
+      System.exit(1);
+    }
+  }
+
+  private void setUp() {
+    channel = createChannel();
+    blockingStub = TestServiceGrpc.newBlockingStub(channel);
+  }
+
+  private void shutdown() {
+    try {
+      if (channel != null) {
+        channel.shutdownNow();
+        channel.awaitTermination(1, TimeUnit.SECONDS);
+      }
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private void run() throws Exception {
+    final long startTimeMillis = System.currentTimeMillis();
```

Comment:
Because `nanoTime()` can become negative, it would require more care when using it. `currentTimeMillis()`'s accuracy and precision is sufficient in this case.

Reply:
To be clear, nanoTime() doesn't necessary start at 0. So it may be negative initially. You must also do the subtraction, but you have to do that anyway.

But I agree that currentTimeMillis() is fine for this task when running on a server, since the clock won't be jumping around.

Reply:
And I'll also mention that nanoTime doesn't take any more effort, since you do the same math independent of whether it is negative. And that's also the same math as used for currentTimeMillis.